### PR TITLE
ci(windows): always cleanup after build or test

### PR DESCRIPTION
**Pull request by ginglis13**
Issue #, if available:

*Description of changes:*

Use the `always()` built in status check function to always run cleanup steps, even if a job is cancelled:
https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions

*Testing done:*

refactor

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by vsiravar** [comment]
Nice, TIL about `always()`. LGTM

**Review by vsiravar — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -207,18 +207,6 @@ jobs:
...
-        run: |
-          wsl --list --verbose
-          wsl --shutdown
-          wsl --unregister lima-finch
-          wsl --list --verbose
-      - name: Clean up previous files
-        run: |
-          Remove-Item C:\Users\Administrator\.finch -Recurse -ErrorAction Ignore
-          Remove-Item C:\Users\Administrator\AppData\Local\.finch -Recurse -ErrorAction Ignore
-          make clean
```
Comment (KevinLiAWS) [inline .github/workflows/ci.yaml:220]: I think make clean still need to execute before build project, I mean this needs to be done both places before and after

Comment (ginglis13) [reply .github/workflows/ci.yaml:220]: Everything should already have been made clean from the `Remove Finch VM and Clean Up Previous Environment` in the previous run

Comment (ginglis13) [reply .github/workflows/ci.yaml:220]: But it's not too much to add and isn't costly, so I'll take your suggestion just to be safe :)

**Review by KevinLiAWS — APPROVED** [review]
LGTM

**Pull request merged**